Code/Framework/AzCore/AzCore/std/ranges/ranges_algorithm.h

```diff
@@ -992,5 +992,69 @@ namespace AZStd::ranges
     {
         constexpr Internal::find_end_fn find_end{};
     }
-} // namespace AZStd::ranges
 
+    namespace Internal
+    {
+        struct all_of_fn
+        {
+            template<class I, class S, class Proj = identity, class Pred,
+                class = enable_if_t<conjunction_v<
+                bool_constant<input_iterator<I>>,
+                bool_constant<sentinel_for<S, I>>,
+                bool_constant<indirect_unary_predicate<Pred, projected<I, Proj>>>>
+            >>
+            constexpr bool operator()(I first, S last, Pred pred, Proj proj = {}) const
+            {
+                return ranges::find_if_not(first, last, std::ref(pred), std::ref(proj)) == last;
```

Comment:
If you include `AzCore/std/reference_wrapper.h` file, the `AZStd::ref` name can be used instead(`AZStd::ref` is just std::ref, but with just bring that name into the AZStd namespace)